src/library/scala/collection/View.scala

```diff
@@ -404,9 +404,15 @@ object View extends IterableFactory[View] {
 
   @SerialVersionUID(3L)
   private[collection] class Patched[A](underlying: SomeIterableOps[A], from: Int, other: IterableOnce[A], replaced: Int) extends AbstractView[A] {
-    def iterator: Iterator[A] = underlying.iterator.patch(from, other.iterator, replaced)
+    // we may be unable to traverse `other` more than once, so we need to cache it if that's the case
+    private val _other: Iterable[A] = other match {
+      case other: Iterable[A] => other
+      case other              => LazyList.from(other)
+    }
+
+    def iterator: Iterator[A] = underlying.iterator.patch(from, _other.iterator, replaced)
     override def knownSize: Int = if (underlying.knownSize == 0 && other.knownSize == 0) 0 else super.knownSize
-    override def isEmpty: Boolean = if (knownSize == 0) true else iterator.isEmpty
+    override def isEmpty: Boolean = if (knownSize == 0) true else _other.isEmpty
```

Comment:
Does this logic work? Would this not be wrong for case 

```scala
List(1,2,3).view.patch(0, Seq(), 0).isEmpty
```
? Looks like with this logic the above would return `true` but in fact it is false.

Reply:
oops, good point. idk why I thought that was a good change. I'm blaming it on me from many months ago

Reply:
funnily enough, the previous line should use `_other` instead of `other` as well